### ISSUE TITLE
Thin space missing after mathtext operators

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#17852]
### Bug report

**Bug summary**

Mathtext operators (`\log`, `\max`, etc.) should insert a thin space after themselves (if followed by ordinary characters, not if followed by an open parenthesis, see https://tex.stackexchange.com/questions/225922/spacing-rules-for-operatorname/225924#225924), but currently it does not.

**Code for reproduction**

```python
from pylab import *
figtext(.5, .6, r"$\log 6$ (usetex)", usetex=True)
figtext(.5, .5, r"$\log 6$ (current)")
figtext(.5, .4, r"$\mathdefault{log}\,6$ (manual thin space)")
figtext(.5, .3, r"$\mathdefault{log}\ 6$ (manual full space)")
show()
```

**Actual outcome**

![out](https://user-images.githubusercontent.com/1322974/86836179-69769900-c09d-11ea-9271-59310788adf5.png)
Note the missing thin space between "g" and "6" in the current behavior.

**Matplotlib version**

  * Operating system: fedora
  * Matplotlib version: master
  * Matplotlib backend (`print(matplotlib.get_backend())`): qt5agg
  * Python version: 38
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by tacaswell** [comment]
Labeled this as hard because the fix is going to involve diving into the tex parser in https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/mathtext.py but good first issue because there is no API design and should not require any changes outside of the `mathtext.py`.

**Comment by jkseppan** [comment]
Is this done since #17890 was merged?

**Comment by anntzer** [comment]
I think sub/superscript handling is still a problem (https://github.com/matplotlib/matplotlib/pull/17890#issuecomment-657160614).

**Comment by oscargus** [comment]
Closed by #23243
